src/item/Item.js

```diff
@@ -2785,11 +2787,27 @@ new function() { // Injection scope for hit-test functions shared with project
      * no children, a {@link TextItem} with no text content and a {@link Path}
      * with no segments all are considered empty.
      *
+     * @param {Boolean} [recursively=false] whether an item with children should
+     * be considered empty if all its descendants are empty
      * @return Boolean
      */
-    isEmpty: function() {
+    isEmpty: function(recursively) {
         var children = this._children;
-        return !children || !children.length;
+        // Item without children is empty.
+        if (!children || !children.length) {
+            return true;
+        // In non-recursive check, item with children is not empty.
+        } else if (!recursively) {
+            return false;
+        // In recursive check, item is empty if all its children are empty.
+        } else {
+            for (var i = 0, l = children.length; i < l; i++) {
+                if (!children[i].isEmpty(true)) {
+                    return false;
+                }
+            }
+            return true;
+        }
```

Comment:
@lehni This PR looks good!

By the way, I have a question about coding style.
This part can remove `else` like,
```javascript
if (!children || !children.length)
    return true;
if (!recursively)
    return false;
for (var i = 0, l = children.length; i < l; i++) {
    if (!children[i].isEmpty(true))
        return false;
}
return true;
```
Which is recommended style in paperjs?

Reply:
I would write it this way:
```js
    isEmpty: function(recursively) {
        var children = this._children;
        var numChildren = children ? children.length : 0;
        if (recursively) {
            for (var i = 0; i < numChildren; i++) {
                if (!children[i].isEmpty(true)) {
                    return false;
                }
            }
        } else if (numChildren) {
            return false;
        }
        return true;
    },
```

But I'm not sure that overloading `isEmpty()` with an optional parameter is the way to go. Maybe we add another method for this check. I need to think about it a little.

Reply:
I got it! Your function looks better.

By the way, when `retursively = true and numChildren = 0`, your function return true ;)
```js
    isEmpty: function(recursively) {
        var children = this._children;
        var numChildren = children ? children.length : 0;
        if (recursively) {
            for (var i = 0; i < numChildren; i++) {
                if (!children[i].isEmpty(true)) {
                    return false;
                }
            }
        }
        return !numChildren;
    },
```